spec/Plugin/ContentTypePluginSpec.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace spec\Http\Client\Common\Plugin;
+
+use PhpSpec\Exception\Example\SkippingException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamInterface;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+
+class ContentTypePluginSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Http\Client\Common\Plugin\ContentTypePlugin');
+    }
+
+    function it_is_a_plugin()
+    {
+        $this->shouldImplement('Http\Client\Common\Plugin');
+    }
+
+    function it_adds_json_content_type_header(RequestInterface $request)
+    {
+        $request->hasHeader('Content-Type')->shouldBeCalled()->willReturn(false);
+        $request->getBody()->shouldBeCalled()->willReturn(json_encode(['foo' => 'bar']));
+        $request->withHeader('Content-Type', 'application/json')->shouldBeCalled()->willReturn($request);
+
+        $this->handleRequest($request, function () {}, function () {});
+    }
+
+    function it_adds_xml_content_type_header(RequestInterface $request)
+    {
+        $request->hasHeader('Content-Type')->shouldBeCalled()->willReturn(false);
+        $request->getBody()->shouldBeCalled()->willReturn('<foo>bar</foo>');
```

Comment:
Can you use a real stream implementation instead of a string ?

I think this test case will fail with a stream implementation as there is no rewind